dpnp/dpnp_utils/dpnp_utils_linearalgebra.py

```diff
@@ -361,13 +381,34 @@ def dpnp_matmul(
         x2_shape = x2.shape
         res_shape = tuple(tmp_shape) + (x1_shape[-2], x2_shape[-1])
 
-    # calculate results
+    # create result array
     result = dpnp.empty(
         res_shape,
         dtype=gemm_dtype,
         usm_type=res_usm_type,
         sycl_queue=exec_q,
     )
+
+    out_is_used = False
+    if out is not None:
+        dpnp.check_supported_arrays_type(out)
```

Comment:
I'd suggest to separate the logic into separate function and to reuse in `dpnp_dot` if possible.
Also it would be help to add description to that new function if any.